client/src/modules/user/UserProfile/EmailNotifications.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { PureComponent } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import AlertBox from 'common/components/AlertBox';
+import { saveEmailNotificationSettings } from '../model/actions';
+import { NotificationFrequency, MessageType } from 'common/constants/enums';
+import SelectDays from './SelectDays';
+import './EmailNotification.scss';
+
+// TODO: Fetch settings when profile view is displayed
+
+class EmailNotifications extends PureComponent {
+  state = {
+    success: false,
+    error: false,
+    notificationFrequency: NotificationFrequency.MONDAY,
+    areDaysVisible: true
+  };
+
+  showDays = () =>
+    this.setState({
+      areDaysVisible: true,
+      notificationFrequency: NotificationFrequency.MONDAY
+    });
+
+  showSuccessMessage = () => {
+    this.setState({ success: true }, this.hideSuccessMessageAfterTimeout);
+  };
+
+  hideSuccessMessageAfterTimeout = () =>
+    setTimeout(() => this.setState({ success: false }), 4000);
+
+  handleSetNever = () =>
+    this.setState(
+      {
+        areDaysVisible: false,
+        notificationFrequency: NotificationFrequency.NEVER
+      },
+      this.updateNotificationSettings
+    );
+
+  handleSelect = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState(
+      { notificationFrequency: value },
+      this.updateNotificationSettings
+    );
+  };
+
+  updateNotificationSettings = async () => {
+    const { notificationFrequency } = this.state;
+
+    this.setState({ error: false });
+
+    try {
+      const result = await saveEmailNotificationSettings(notificationFrequency);
+
+      if (result) {
+        this.showSuccessMessage();
+      }
+    } catch {
+      this.setState({ error: true });
```

Comment:
we display such errors in our notifications (excluding forms where we want to display errors inline). I don't say that your solution is wrong, maybe it's even better but now it's patchy also